### DISCUSSION TITLE
Custom history of uptime instead of 7 days

### DISCUSSION THREAD

**Discussion by vigneshksaithal** [upptime/upptime#80] [Ideas] [answered]

**Comment by AnandChowdhary** [comment]
Sounds like a good idea, we'll have to explore the best way to implement this without breaking the GitHub API public rate limit.

(I'm assuming you mean the 7-day response time graph: https://demo.upptime.js.org/history/google. Please correct me if I'm wrong!)

**Comment by vigneshksaithal** [comment]
Yes, I meant 7-day reponse time only.

**Comment by AnandChowdhary** [comment]
Upptime now adds all these different types of uptime metrics:

- Overall uptime
- 1-day (24h) uptime
- 1-week (7d) uptime
- 1-month uptime
- 1-year uptime

Soon, I'll also add all of these types for response time metrics.

**Comment by AnandChowdhary** [comment]
Response time also has these metrics now! 😄

All that's left to do is add the companion interface for the web app.

**Comment by AnandChowdhary** [comment, marked as answer]
The status website now has support for all the metrics in all the durations! 😄

![Screen Shot 2020-12-06 at 14 47 09](https://user-images.githubusercontent.com/2841780/101276305-f7cc8380-37d1-11eb-8ada-88d69832120c.png)

The README.md file also has them, like this:

![Screen Shot 2020-12-06 at 14 49 17](https://user-images.githubusercontent.com/2841780/101276391-4e39c200-37d2-11eb-8953-a011052a6f9c.gif)

**Comment by vigneshksaithal** [comment]
Great work. 👍
